src/main/java/hudson/plugins/audit_trail/ElasticSearchAuditLogger.java

```diff
@@ -347,23 +348,31 @@ private CloseableHttpClient createHttpClient(
             }
             SSLContext sslContext = contextBuilder.build();
             HttpClientBuilder builder = HttpClients.custom();
-            builder.setSslcontext(sslContext);
+
+            SSLConnectionSocketFactoryBuilder sslConnectionBuilder =
+                    SSLConnectionSocketFactoryBuilder.create().setSslContext(sslContext);
             if (skipCertificateValidation) {
-                builder.setSSLHostnameVerifier(new NoopHostnameVerifier());
+                sslConnectionBuilder.setHostnameVerifier(NoopHostnameVerifier.INSTANCE);
             }
+
+            builder.setConnectionManager(PoolingHttpClientConnectionManagerBuilder.create()
+                    .setSSLSocketFactory(sslConnectionBuilder.build())
+                    .build());
+
             return builder.build();
         }
 
         public void sendMessage(String event) throws IOException {
             HttpPost post = getHttpPost(event);
-            try (CloseableHttpResponse response = httpClient.execute(post)) {
-                int statusCode = response.getStatusLine().getStatusCode();
+            httpClient.execute(post, response -> {
```

Comment:
Needs some testing, behavior of the method changed, it was previously synchronous.
(note for myself :)).

Reply:
There was a slight issue with the logging of a communication error (fixed now).